### ISSUE TITLE
building js9 from 9.0.3 branch fails

### ISSUE THREAD

**Issue by yveskerwyn** [Jumpscale/developer#56]
After having removed all containers and images:
```
eval "$(ssh-agent -s)"
ssh-add -K ~/.ssh/id_rsa
export GIGDIR="/Users/$USER/gig9"
rm -rf $GIGDIR
export GIGBRANCH="9.0.3"
export GIGSAFE=1
rm -rf /tmp/jsinit.sh
curl https://raw.githubusercontent.com/Jumpscale/developer/${GIGBRANCH}/jsinit.sh?$RANDOM > /tmp/jsinit.sh
bash /tmp/jsinit.sh
source ~/.jsenv.sh
js9_build -l -p
```

Here's the result:
```
...
Found existing installation: cryptography 1.2.3
Uninstalling cryptography-1.2.3:
Successfully uninstalled cryptography-1.2.3
Found existing installation: paramiko 1.16.0
Uninstalling paramiko-1.16.0:
Successfully uninstalled paramiko-1.16.0
Found existing installation: psutil 3.4.2
E:DEPRECATION: Uninstalling a distutils installed project (psutil) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
Uninstalling psutil-3.4.2:
Successfully uninstalled psutil-3.4.2
Running setup.py develop for JumpScale9Lib
Complete output from command /usr/bin/python3 -c "import setuptools, tokenize;__file__='/opt/code/github/jumpscale/lib9/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps:
running develop
running egg_info
writing requirements to JumpScale9Lib.egg-info/requires.txt
writing dependency_links to JumpScale9Lib.egg-info/dependency_links.txt
writing top-level names to JumpScale9Lib.egg-info/top_level.txt
writing JumpScale9Lib.egg-info/PKG-INFO
reading manifest file 'JumpScale9Lib.egg-info/SOURCES.txt'
writing manifest file 'JumpScale9Lib.egg-info/SOURCES.txt'
running build_ext
Creating /usr/local/lib/python3.5/dist-packages/JumpScale9Lib.egg-link (link to .)
Adding JumpScale9Lib 9.0.3 to easy-install.pth file

Installed /opt/code/github/jumpscale/lib9
Running post install task
[Wed28 21:21] - SystemProcess.py    :661 :j.sal.process                  - INFO     - Checking whether at least 1 processes redis-server are running
Collecting 0-core-client from git+https://github.com/zero-os/0-core#egg=0-core-client&subdirectory=client/py-client
Cloning https://github.com/zero-os/0-core to /tmp/pip-build-caothk_2/0-core-client
Requirement already satisfied: redis>=2.10.5 in /usr/local/lib/python3.5/dist-packages (from 0-core-client)
Installing collected packages: 0-core-client
Running setup.py install for 0-core-client: started
Running setup.py install for 0-core-client: finished with status 'done'
Successfully installed 0-core-client-1.1.0a4
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/opt/code/github/jumpscale/lib9/setup.py", line 98, in <module>
'developement': develop
File "/usr/lib/python3.5/distutils/core.py", line 148, in setup
dist.run_commands()
File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
self.run_command(cmd)
File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
cmd_obj.run()
File "/opt/code/github/jumpscale/lib9/setup.py", line 37, in run
self.execute(_post_install, (libname, libpath), msg="Running post install task")
File "/usr/lib/python3.5/distutils/cmd.py", line 336, in execute
util.execute(func, args, msg, dry_run=self.dry_run)
File "/usr/lib/python3.5/distutils/util.py", line 301, in execute
func(*args)
File "/opt/code/github/jumpscale/lib9/setup.py", line 19, in _post_install
j.tools.jsloader.generateJumpscalePlugins()
AttributeError: 'JSLoader' object has no attribute 'generateJumpscalePlugins'
'JSLoader' object has no attribute 'generateJumpscalePlugins'

----------------------------------------
E:Command "/usr/bin/python3 -c "import setuptools, tokenize;__file__='/opt/code/github/jumpscale/lib9/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps" failed with error code 1 in /opt/code/github/jumpscale/lib9/
Traceback (most recent call last):
  File "/usr/local/bin/js9_getcode_libs_prefab_ays", line 26, in <module>
    j.do.execute("cd %s;sh install.sh" % path)
  File "/opt/code/github/jumpscale/core9/JumpScale9/core/InstallTools.py", line 1986, in execute
    die=die)
  File "/opt/code/github/jumpscale/core9/JumpScale9/tools/executor/ExecutorLocal.py", line 41, in execute
    outputStderr=outputStderr, timeout=timeout)
  File "/opt/code/github/jumpscale/core9/JumpScale9/sal/process/SystemProcess.py", line 203, in execute
    raise RuntimeError("Could not execute cmd:\n'%s'\nerr:\n%s" % (command, err))
RuntimeError: Could not execute cmd:
'set -e;cd /opt/code/github/jumpscale/lib9;sh install.sh'
err:
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
dpkg-preconfigure: unable to re-open stdin:
    DEPRECATION: Uninstalling a distutils installed project (psutil) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
Command "/usr/bin/python3 -c "import setuptools, tokenize;__file__='/opt/code/github/jumpscale/lib9/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps" failed with error code 1 in /opt/code/github/jumpscale/lib9/
Could not execute cmd:
'set -e;cd /opt/code/github/jumpscale/lib9;sh install.sh'
err:
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
dpkg-preconfigure: unable to re-open stdin:
    DEPRECATION: Uninstalling a distutils installed project (psutil) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
Command "/usr/bin/python3 -c "import setuptools, tokenize;__file__='/opt/code/github/jumpscale/lib9/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps" failed with error code 1 in /opt/code/github/jumpscale/lib9/
```

**Comment by ahussein** [comment]
Up until version js9 version 9.0.3 we had a problem the install scripts do not consider the GIGBRANCH variable when cloning out repository inside the docker image (only the core9 repo is taking into consideration that variable). This means that while you are exporting GIGBRANCH=9.0.3 the code for libs9  repo will come from master branch and that is the reason for the error you are seeing.

Now on master branch for developer repo, the GIGBRANCH variable is considered and passed to the scripts that clone the other repositories.

I can see the following solutions for your problem:

-  Do the same steps as you mentioned but also export the variable GIGDEVELOPERBRANCH=master as mentioned in https://github.com/Jumpscale/developer , This way you will install from branch 9.0.3 for all the code repos and only the installer scripts will come from master branch of developer repo.

- Install from master branch by exporting both variables GIGBRANCH and GIGDEVELOPERBRANCH to master

- We backport the fixes on developer master branch to 9.0.3 branch and then you can install from 9.0.3 for all repositories.

**Comment by rkhamis** [comment]
> We backport the fixes on developer master branch to 9.0.3 branch and then you can install from 9.0.3 for all repositories.

Yes, that's what we should do
